src/cleo/io/inputs/argv_input.py

```diff
@@ -43,20 +43,18 @@ def first_argument(self) -> str | None:
         is_option = False
 
         for i, token in enumerate(self._tokens):
-            if token and token[0] == "-":
-                if token.find("=") != -1 or len(self._tokens) <= (i + 1):
+            if token.startswith("-"):
+                if "=" in token or len(self._tokens) == (i + 1):
                     continue
 
                 # If it's a long option, consider that
                 # everything after "--" is the option name.
                 # Otherwise, use the last character
                 # (if it's a short option set, only the last one
                 # can take a value with space separator).
-                name = token[2:] if len(token) > 1 and token[1] == "-" else token[-1]
```

Comment:
> Why the change from <= to ==?

Checking if the length of the list we're iterating over is _less_ than the current index is slightly confusing and would have sense if the list did change it's size during the loop. It doesn't.

> And why was and token[1] == "-" removed?

That was a mistake, sorry. My bad. Thanks for spotting that. To avoid any further confusion, I've made the conditional check more readable..